src/js/services/integrations-service.js

```diff
@@ -37,17 +38,55 @@ export function resetTwilioAttributes() {
 export function fetchTwilioAttributes() {
     const {user: {clients, pendingClients}} = store.getState();
     const client = clients.find((client) => client.platform === 'twilio');
-    const pendingClient = pendingClients.find((client) => client.platform === 'twilio');
+    const pendingClient = pendingClients && pendingClients.find((client) => client.platform === 'twilio');
 
     if (client) {
         updateTwilioAttributes({
             linkState: 'linked',
-            number: client.displayName
+            appUserNumber: client.displayName
         });
     } else if (pendingClient) {
         updateTwilioAttributes({
             linkState: 'pending',
-            number: pendingClient.displayName
+            appUserNumber: pendingClient.displayName
         });
     }
 }
+
+export function linkTwilioChannel(userId, data) {
+    return core().appUsers.link.linkChannel(userId, data)
+        .then((appUser) => {
+            immediateUpdate(appUser).then(() => {
+                updateTwilioAttributes({
+                    linkState: 'pending'
```

Comment:
I might update the `appUserNumber` here to match what's returned (say we validate the number but store a slightly different format)

I'm not sure if this will be applicable though, just worth mentioning.